src/main/java/com/mongodb/jdbc/MongoSQLStatement.java

```diff
@@ -31,10 +34,30 @@ public ResultSet executeQuery(String sql) throws SQLException {
                 iterable = iterable.batchSize(fetchSize);
             }
 
-            resultSet = new MongoSQLResultSet(this, iterable.cursor());
+            MongoIterable<MongoJsonSchemaResult> schemaIterable =
+                    currentDB
+                            .withCodecRegistry(MongoDriver.registry)
+                            .aggregate(
+                                    Collections.singletonList(schemaStage),
+                                    MongoJsonSchemaResult.class)
+                            .maxTime(maxQuerySec, TimeUnit.SECONDS);
+            MongoJsonSchemaResult schemaResult = schemaIterable.cursor().next();
+            if (schemaResult.ok != 1) {
+                throw new SQLException("Invalid schema result from server");
```

Comment:
We may want to ask the ADL team if they return an `err` or `error` field or something that we could pass along to users to provide more details.

Reply:
@henretty This is where we're calling `sqlGetResultSchema`. 2 questions:
1. Is it possible for this to return a result with `ok` != 1?
2. If so, does the result have any error information we can/should share with JDBC users?

Reply:
When running the runCommand({sqlGetResultSchema:1, ...}) against ADL it hit an error that threw an exception at the runCommand() line, prior to this check of schemaResult.ok !=1.  There is a check done that throws the error if the response for ok is not 1, [here.](https://github.com/mongodb/mongo-java-driver/blob/53b0ff156a33d883ec6c78a64d2c9c3d14e9218c/driver-core/src/main/com/mongodb/internal/connection/InternalStreamConnection.java#L417)

So it seems that by the time we get to line 45, it has already been checked making it no longer necessary.  Does that sound correct?

Reply:
Ah that sounds good to me. It's weird they throw an exception for that considering mongo has pretty rich/detailed error codes, but I suppose exceptions are the java-way. Given that check, there is no need for us to do the `ok != 1` check ourselves.

Do you think we should be catching and handling that exception, though?

Reply:
It throws a runtime exception, which leads me to think the error that is thrown is meant to stop execution and show the error.  In terms of handling there isn't much to do if the schema wasn't returned, besides retry if it were due to a timeout perhaps.  
I'm leaning towards leaving it for now.

Reply:
Makes sense. Sounds good to me 👍